### DISCUSSION TITLE
Setting up https For Open-Webui using Nginx Proxy Manager

### DISCUSSION THREAD

**Discussion by rootcli** [open-webui/open-webui#3028] [Show and tell]
2024-06-11

## Setting up https For Open-Webui

My Current Setup: 
- Ollama installed locally
- Open-Webui installed on Docker
- [Nginx Proxy Manager](https://github.com/NginxProxyManager/nginx-proxy-manager) installed on Docker

### Step 1: Create a Self-Signed Certificate

> [!note]
> **Here we will use *ai.local* as a domain example, change it to your local domain**

1. **Generate the Certificate and Key**:

```bash
mkdir -p ~/certs 
openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout ~/certs/selfsigned.key -out ~/certs/selfsigned.crt -subj "/CN=[ai.local]"
```

### Step 2: Configure Nginx Proxy Manager to Use the Self-Signed Certificate

1. **Upload the Self-Signed Certificate**:
    
    - Log in to Nginx Proxy Manager.
    - Go to the `SSL Certificates` tab.
    - Click `Add SSL Certificate` and select `Custom`.
    - Fill in the certificate details:
        - Name: `YOUR-LOCAL-DOMAIN self-signed`
        - Certificate: Copy the contents of `~/certs/selfsigned.crt`.
        - Certificate Key: Copy the contents of `~/certs/selfsigned.key`.
    - Click `Save`.

1. **Add a New Proxy Host**:
    
    - Go to the `Proxy Hosts` tab.
    - Click `Add Proxy Host`.
    - Enter `ai.local` in the `Domain Names` field.
    - Set the `Scheme` to `http`.
    - Enter `localhost` in the `Forward Hostname / IP` field and `PORT-NUMBER-HERE` in the `Forward Port` field.
    - Check `Block Common Exploits`.
    - In the `SSL` tab, select `ai.local self-signed` from the `SSL Certificate` dropdown.
    - Check `Force SSL`.
    - Click `Save`.

### Step 3: Update Your Hosts File

Ensure that `ai.local` resolves to your local machine by updating the `/etc/hosts` file:

`sudo nano /etc/hosts`

Add the following line:

`127.0.0.1 ai.local`

### Step 4: Restart Nginx Proxy Manager

Restart the Nginx Proxy Manager to ensure all configurations are applied:

`docker-compose restart`

### Step 5: Verify the Setup

1. **Open your browser** and navigate to `https://ai.local`.
2. **Verify** that the site loads correctly and is served over HTTPS.

> [!note]
> You might see a warning in your browser indicating that the certificate is self-signed. You can safely ignore this warning for local development.

##### Hope this help and good luck

**Comment by czar** [comment]
This was helpful, thank you. Everything worked A-okay except that WebSocket support needed to be checked (enabled), else Ollama reports a "network error". Also, note that the "forward port" in my case was 3000.

**Comment by vilemj-Viclick** [comment]
This was really helpful just one more note for dumbasses like me:

## Make sure you read properly which field is the key and which is the certificate. ;-)

**Comment by hdnh2006** [comment]
Hello community! 

I have created an article on my blog talking about how to deploy Open WebUI in https:
https://henrynavarro.org/deploy-your-own-open-webui-interface-with-https-security-53a6ea2609d7?sk=a5876acd56b44ea60f10f9b13aa24aee

And also I have a YouTube video (in spanish 🇪🇸) talking about it:
https://youtu.be/CcGrI9YkUJI?si=d-cskZZTwEdz8Qsv

I hope this could be useful for you.

Enjoy!
